Result/ResultType.swift

```diff
@@ -65,9 +71,9 @@ public extension ResultType {
 }
 
 /// Protocol used to constrain `tryMap` to `Result`s with compatible `Error`s.
-public protocol ErrorTypeConvertible: ErrorType {
+public protocol ErrorTypeConvertible: ResultErrorType {
 	typealias ConvertibleType = Self
-	static func errorFromErrorType(error: ErrorType) -> ConvertibleType
+	static func errorFromErrorType(error: ResultErrorType) -> Self
```

Comment:
What is the build failure? And if that is the case, `ConvertibleType` should not be needed anymore.